Failed to generate build files on clang GNU frontend on Windows with `-DJPEGXL_STATIC=ON`

**Describe the bug**
After #3837, CMake can no longer produce build files due to failing atomics check. This only happens with Clang with GNU driver and `-DJXL_STATIC=ON`. This means, MSVC and Clang with MSVC driver (clang-cl) are not affected, and Clang with GNU driver with `-DJXL_STATIC=OFF` is not affected.
```
-- The C compiler identification is Clang 19.1.0 with GNU-like command-line
-- The CXX compiler identification is Clang 19.1.0 with GNU-like command-line
...
-- Performing Test ATOMICS_IN_LIBRARY
-- Performing Test ATOMICS_IN_LIBRARY - Failed
CMake Error at cmake/FindAtomics.cmake:49 (message):
  Neither lock free instructions nor -latomic found.
Call Stack (most recent call first):
  C:/Users/Alifian/scoop/apps/vcpkg/current/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:262 (find_package)
```

**To Reproduce**
Steps to reproduce the behavior:
Use `-DJPEGXL_STATIC=ON` together with Clang-GNU driver (`-DCMAKE_CXX_COMPILER=clang++`)

**Environment**
 - OS: [e.g. Windows 11 23H2]
 - Compiler version: Clang 19.1.0
 - CPU type: x86_64

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,7 +225,9 @@ if(JPEGXL_STATIC)
 
   # https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
   # https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html
-  set(CMAKE_MSVC_RUNTIME_LIBRARY "$<$<NOT:$<CONFIG:Debug>>:MultiThreaded>" CACHE STRING "")
+  # For debug builds we intentionally link with shared library to ensure that we don’t accidentally
+  # redistribute such binaries anywhere.
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:DebugDLL>" CACHE STRING "")
 
   # Clang developers say that in case to use "static" we have to build stdlib
   # ourselves; for real use case we don't care about stdlib, as it is "granted",

